x2js.js

```diff
@@ -537,7 +543,7 @@
 		function serializeJavaScriptObject(element, elementName, attributes) {
 			var result = "";
 
-			if (element === undefined || element === null || element === '') {
+			if (element === undefined || element === null || element === '' && config.selfClosingElements) {
```

Comment:
The && operator has higher precedence so the new configuration option will only affect the empty string case. I presume this is unintentional. A "Element with attribute and selfClosingElements set to true" test case would be perfect to catch this.